# Lecture #4
<br>
Arithmetic and Comparison Operators (C vs C++ vs Python)

# Arithmetic Operators (C vs C++ vs Python)
<br>
Arithmetic operators perform mathematical calculations such as addition, subtraction, multiplication, division, and finding remainders.



## Common Arithmetic Operators

| Operation | C | C++ | Python |
|------------|---|-----|--------|
| Addition | `+` | `+` | `+` |
| Subtraction | `-` | `-` | `-` |
| Multiplication | `*` | `*` | `*` |
| Division | `/` | `/` | `/` |
| Modulus (Remainder) | `%` | `%` | `%` |
| Exponent (Power) | Not available directly | Not available directly | `**` |
| Floor Division | Not available directly | Not available directly | `//` |

## Key Points

- Most arithmetic operators have the same symbols in **C, C++, and Python**.
- Python provides additional operators such as:
  - `**` for exponentiation.
  - `//` for floor division.
- C and C++ use the `pow()` function from the **math library** for exponentiation.
- Operator precedence is generally similar across all three languages.
- Parentheses `()` can be used to control the order of evaluation.


## Examples

### 1. Addition

### C/C++

```cpp
#include <iostream>
using namespace std;

int main() {
    int a = 10, b = 3;
    cout << a + b;
}
```


### Python

In [1]:

a = 10
b = 3

print(a + b)

13


### 2. Division Difference

### C/C++

```cpp
#include <iostream>
using namespace std;

int main() {
    int a = 10, b = 3;
    cout << a / b;
}
```


### Python

In [2]:
a = 10
b = 3

print(a / b)

3.3333333333333335


> **Note:** In Python, the `/` operator always returns a floating-point value.

### 3. Floor Division

In [3]:
print(10 // 3)   # 3
print(11 // 3)   # 3


3
3


### 4. Exponentiation

### C/C++

```cpp
#include <cmath>

pow(2, 3);
```


### Python


In [4]:
print(2 ** 3)

8


# Comparison (Relational) Operators
<br>
Comparison operators compare two values and return a Boolean result.

- **Python:** `True` or `False`
- **C:** `1` or `0`
- **C++:** `true` or `false`


## Common Comparison Operators

| Meaning | C | C++ | Python |
|---------|---|-----|--------|
| Equal To | `==` | `==` | `==` |
| Not Equal To | `!=` | `!=` | `!=` |
| Greater Than | `>` | `>` | `>` |
| Less Than | `<` | `<` | `<` |
| Greater Than or Equal To | `>=` | `>=` | `>=` |
| Less Than or Equal To | `<=` | `<=` | `<=` |

## Key Points

- Comparison operators are almost identical in C, C++, and Python.
- They return:
  - `True` or `False` in Python.
  - `1` or `0` in traditional C.
  - `true` or `false` in modern C++.
- These operators are commonly used in:
  - `if` statements
  - `while` loops
  - Conditional expressions
- Do **not** confuse:
  - Assignment (`=`)
  - Comparison (`==`)


## Examples

### 1. Equality Check

### C++

```cpp
int x = 5;

if (x == 5)
    cout << "Equal";
```

### Python

In [5]:
x = 5

if x == 5:
    print("Equal")

Equal


### 2. Greater Than

### C++

```cpp
cout << (10 > 3);
```

### Python

In [6]:
print(10 > 3)

True


### 3. Not Equal To

### C++

```cpp
cout << (10 != 5);
```


### Python

In [7]:
print(10 != 5)

True


# Important Differences Between C/C++ and Python

| Feature | C/C++ | Python |
|---------|--------|--------|
| `10 / 3` | `3` (Integer Division) | `3.3333333333333335` |
| Floor Division | Not available directly | `//` |
| Exponentiation | `pow()` | `**` |
| Boolean Output | `1/0` or `true/false` | `True/False` |
| Syntax | More verbose | More concise |

# Quick Classroom Summary

- Arithmetic operators perform mathematical calculations.|
- Comparison operators compare values and return Boolean results.
- Most operators have identical symbols in C, C++, and Python.
- Python introduces:
  - `**` for exponentiation.
  - `//` for floor division.
- Python's `/` always returns a floating-point result.
- Comparison operators are fundamental for decision-making and loops.

## Takeaway

> **Remember:**
>
> - Use `+`, `-`, `*`, `/`, and `%` for basic arithmetic.
> - Python provides two extra arithmetic operators:
>   - `//` → Floor Division
>   - `**` → Exponentiation
> - Use comparison operators (`==`, `!=`, `>`, `<`, `>=`, `<=`) to compare values.
> - Comparison operators return Boolean values, making them essential for `if` statements and loops.

# Thank You!

# Class Tasks...

# Task 1

Given the following two variables:

```python
x = 15
y = 10
```

Write Python expressions to check the following:

1. Is `x` greater than `y`?
2. Is `x` less than `y`?
3. Are `x` and `y` equal?
4. Are `x` and `y` not equal?
5. Is `x` greater than or equal to `y`?
6. Is `x` less than or equal to `y`?

Print the result of each comparison.

# Solution 1

In [8]:
x = 15
y = 10

print("x > y:", x > y)
print("x < y:", x < y)
print("x == y:", x == y)
print("x != y:", x != y)
print("x >= y:", x >= y)
print("x <= y:", x <= y)

x > y: True
x < y: False
x == y: False
x != y: True
x >= y: True
x <= y: False


# Class Task 2

A temperature is given in Celsius:

```python
celsius = 35
```

Convert the temperature from Celsius to Fahrenheit using the following formula:

$$
F = (C \times \frac{9}{5}) + 32
$$

Write a Python program that:

1. Stores the Celsius temperature in a variable.
2. Calculates the Fahrenheit temperature.
3. Displays both temperatures.

# Solution 2

In [9]:
celsius = 35

fahrenheit = (celsius * 9 / 5) + 32

print("Temperature in Celsius:", celsius)
print("Temperature in Fahrenheit:", fahrenheit)

Temperature in Celsius: 35
Temperature in Fahrenheit: 95.0


# Class Task 3

A student buys the following items:

```python
notebook_price = 120
pen_price = 50
notebooks = 3
pens = 4
```

Write a Python program to calculate:

1. The total cost of the notebooks.
2. The total cost of the pens.
3. The total bill.
4. Whether the total bill is greater than `500`.

# Solution 3

In [10]:
notebook_price = 120
pen_price = 50
notebooks = 3
pens = 4

notebook_cost = notebook_price * notebooks
pen_cost = pen_price * pens
total_bill = notebook_cost + pen_cost

print("Notebook Cost:", notebook_cost)
print("Pen Cost:", pen_cost)
print("Total Bill:", total_bill)
print("Bill greater than 500:", total_bill > 500)

Notebook Cost: 360
Pen Cost: 200
Total Bill: 560
Bill greater than 500: True
